docs/adrs/0076-partial-update-patches.md

```diff
@@ -0,0 +1,146 @@
+	BillableWeightCap                *unit.Pound
+	BillableWeightJustification      *string
+	NTSRecordedWeight                *unit.Pound
+	ShipmentType                     MTOShipmentType
+	Status                           MTOShipmentStatus
+	Diversion                        bool
+	RejectionReason                  *string
+	Distance                         *unit.Miles
+	Reweigh                          *Reweigh
+	UsesExternalVendor               bool
+	StorageFacility                  *StorageFacility
+	StorageFacilityID                *uuid.UUID
+	ServiceOrderNumber               *string
+	TACType                          *LOAType
+	SACType                          *LOAType
+	PPMShipment                      *PPMShipment
+	CreatedAt                        time.Time
+	UpdatedAt                        time.Time
+	DeletedAt                        *time.Time
+}
+```
+
+Handlers would be in charge of converting payloads into the MTOShipmentUpdate model and passing it to the UpdateShipment service. The UpdateShipment service would be responsible for fetching the database model (MTOShipment) and updating it appropriately based on the `Present` fields in the MTOShipmentUpdate.
+
+Such a design would have many advantages in addition to solving the problem of partial model updates and setting fields to null. The UpdateMTOShipment struct provides clarity by broadcasting intent: it clearly exists to update a shipment and it tells the service exactly what it wants. It also helps to further disentangle the handler layer from the database and services; since the handler now uses the MTOShipmentUpdate model, it no longer needs to know about the database structure. This design will also allow for a thorough cleanup of a lot of the handler and service logic through removal of repeated null checking and shifting business logic responsibilities cleanly to the services.
+
+### Implementation
+
+Due to the scope of work required, implementating this pattern would need to be done slowly.
+
+#### Decisions to finalize
+* Decide on a naming convention for intermediary update models
+* Determine if all models should fall under this pattern or only update models.
+
+#### Refactor stages
+* Identify and refactor any patch endpoints that require nullable fields for deleting values.
+* Fully refactor related validator methods to cleanly handle intermediary update models
+* Refactor all update models to remove dependencies on the `models` package
+* Relocate fully refactored update models into a separate package for better separation of code
+
+
+## Considered Alternatives
+* *Add a new changed values criterion for distinguishing between events when matching to event templates*
+* *Refactor our details types and details components so that different data can be displayed uniquely within the same details type*
+* *Do nothing*
+
+## Decision Outcome
+
+### Chosen Alternative: *Introduce a new struct that serves between the handler and service layers and broadcasts intent*
```

Comment:
**request:** I'd like to hear more about what the pros and cons are as it relates to other parts of the application that aren't move history.

Reply:
Oops, this was a copy/paste error. Fixed!